### PR TITLE
init OperateDataHeader

### PR DESCRIPTION
**Pull request by Jazz86**

### CODE REVIEW

File: src/Header/OperateData.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    std::string WWWAuthenticate;
+};
+class OperateData
+{
+public:
+    OperateData();
+    OperateData(OperateData &right);
+    RequestHeader requestHeader;
+    ResponseHeader responHeader;
+    std::unordered_map<std::string, std::string> body;
```
Comment (john0312) [inline src/Header/OperateData.h]: request body? What about response?

File: src/Header/OperateData.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#ifndef GROUP4_SRC_OPERATEDATA_H_
+#define GROUP4_SRC_OPERATEDATA_H_
+
+#include <string>
+#include <unordered_map>
+
+// This class is a self-defined data structure containing header and body.
+// OperateData newData = OperateData();
+// OperateData["key"] = "value";
+struct RequestHeader
```
Comment (john0312) [inline src/Header/OperateData.h]: Consider using unordered_map for this

File: src/Header/OperateData.h (hunk 1 above)
Comment (Jazz86) [reply src/Header/OperateData.h]: both header and body are store in format of unordered_map, user should choose to use as request data or response data

File: src/Header/OperateData.h (hunk 2 above)
Comment (Jazz86) [reply src/Header/OperateData.h]: done.

File: src/Header/OperateData.h (hunk 1 above)
Comment (john0312) [reply src/Header/OperateData.h]: How do they know which entry of header/body belongs to the response or the request? Do you want to separate them?

Comment (Jazz86) [reply src/Header/OperateData.h]: One OperateData is only used as request data or response data. For example, if it used as request data, then the header in the OperateData only contain request header same as the body. So, the user should be aware of the naming of the OperateData to distinguish between request or response data.

File: src/Header/OperateData.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+// This class is a self-defined data structure containing header and body.
+// User should choose to use only as request data or response data
+// OperateData newData = OperateData();
+// OperateData["key"] = "value";
+class OperateData {
+ public:
+  OperateData();
+  OperateData(OperateData &right);
+  std::unordered_map<std::string, std::string> header;
+  std::unordered_map<std::string, std::string> body;
```
Comment (john0312) [inline src/Header/OperateData.h:23]: Please add variable comment.

File: src/Header/OperateData.h [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+// This class is a self-defined data structure containing header and body.
+// User should choose to use only as request data or response data
+// OperateData newData = OperateData();
+// OperateData["key"] = "value";
+class OperateData {
+ public:
+  OperateData();
+  OperateData(OperateData &right);
+  std::unordered_map<std::string, std::string> header;
```
Comment (john0312) [inline src/Header/OperateData.h:21]: Please add comment or variable on where data other than header or body is.
i.e. HTTP Request Method, Request URL, and Response Status Code.

File: src/Header/OperateData.h (hunk 3 above)
Comment (Jazz86) [reply src/Header/OperateData.h:23]: done.

File: src/Header/OperateData.h (hunk 4 above)
Comment (Jazz86) [reply src/Header/OperateData.h:21]: done.

File: src/Header/OperateData.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ public:
+  OperateData();
+  OperateData(OperateData &right);
+  // Besides HTTP headers, HTTP request method, Request URL, Status Code
+  // would store in the header variable.
+  std::unordered_map<std::string, std::string> header;
+  // Only contain HTTP body.
+  std::unordered_map<std::string, std::string> body;
+  // only used to access body data
+  std::string &operator[](std::string bodyKey);
```
Comment (younger15) [inline src/Header/OperateData.h]: does this parameter need to use reference?

**Review by john0312 — APPROVED** [review]
Approved, you are free to merge this PR.

**Pull request merged**